fullrt.go

```diff
@@ -0,0 +1,629 @@
+package zikade
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/ipfs/boxo/provider"
+	"github.com/ipfs/go-cid"
+	ds "github.com/ipfs/go-datastore"
+	record "github.com/libp2p/go-libp2p-record"
+	recpb "github.com/libp2p/go-libp2p-record/pb"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/peerstore"
+	"github.com/libp2p/go-libp2p/core/routing"
+	"github.com/multiformats/go-multiaddr"
+	mh "github.com/multiformats/go-multihash"
+	"go.opentelemetry.io/otel/attribute"
+	otel "go.opentelemetry.io/otel/trace"
+	"golang.org/x/exp/slog"
+
+	"github.com/plprobelab/zikade/internal/coord"
+	"github.com/plprobelab/zikade/internal/coord/coordt"
+	"github.com/plprobelab/zikade/internal/coord/query"
+	"github.com/plprobelab/zikade/kadt"
+	"github.com/plprobelab/zikade/pb"
+)
+
+type FullRT struct {
```

Comment:
It doesn't make much sense to me to call this `FullRT`. It seems to just be following the same naming as the go-libp2p-kad-dht implementation.

It seems to me that this is really a specialised routing table population strategy for the DHT. Can we make it an option on the normal DHT type?